src/app/shared/validators/min-max-validate.directive.ts

```diff
@@ -23,13 +26,19 @@ export class MinMaxValidateDirective implements Validator {
   }
 
   validate(control: AbstractControl): ValidationErrors | null {
-    if (Validators.required(control)) {
+    this.getSettings();
+    const error: ValidationErrors = {};
+    const value = control.value ? control.value.replace(DIGIT_REGEX, '') : null;
+    const valueNumber = Number(value);
+
+    if (!value && !this.required) {
       return null;
     }
 
-    const error: ValidationErrors = {};
-    const value = control.value.replace(DIGIT_REGEX, '');
-    const valueNumber = Number(value);
+    if (!value && this.required) {
+      error[ErrorCode.required] = {};
+      return error;
+    }
```

Comment:
Do we have to do it all manually? Don't we already have built-in validators for such things that we can (& should) just use?

Reply:
Tak jest wbudowany walidator required, ale uruchamia się przed usunięciem niedozwolonych znaków. 
Uruchamiał przed tym @HostListener('input', ['$event']) onInputChange(event: InputEvent) {...
I to powodowało problemy stąd własny  parametr required